components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1249,22 +1272,32 @@ private class MonitoredItemWrapper {
                 }
                 while (superType != null);
 
-                var propertyNames = new List<QualifiedName>();
+                var fieldNames = new List<QualifiedName>();
                 foreach (var node in nodes) {
-                    foreach (var reference in node.ReferenceTable) {
-                        if (reference.ReferenceTypeId == ReferenceTypeIds.HasProperty) {
-                            propertyNames.Add(nodeCache.FetchNode(reference.TargetId).BrowseName);
-                        }
-                    }
+                    ParseFields(nodeCache, fieldNames, node);
```

Comment:
Any reason for not returning the list from `ParseFields` instead? Why is it an argument?

Reply:
Efficiency reasons, not having to copy a range of items all the time.